pkg/k8s/identitybackend/identity.go

```
@@ -308,8 +315,19 @@ func (c *crdBackend) Release(ctx context.Context, id idpool.ID, key allocator.Al
 	return nil
 }
 
+func GetIdentitiesByKeyFunc(keyType allocator.AllocatorKey) func(obj interface{}) ([]string, error) {
+	return func(obj interface{}) ([]string, error) {
+		if identity, ok := obj.(*v2.CiliumIdentity); ok {
+			return []string{keyType.PutKeyFromMap(identity.SecurityLabels).GetKey()}, nil
+		}
+		return []string{}, nil
```

Seems like we should return an error here.

---

I'm not sure. I tried to find some examples from kubernetes but couldn't.
Basically if we ever return error here the store would panic.

I'll come back to this.

---

It's "impossible", because it would mean we put something wrong in the store. Convention is to log **something** just to catch coding errors. It doesn't have to be sophisticated.

---

Fixed.